### PR TITLE
DelayedClaimable contract extension of Claimable fix #64

### LINKED ISSUE #64: Make a version of Claimable that requires passing of time to confirm new ownership
Idea from: http://vessenes.com/we-need-some-best-practices-for-smart-contracts/

> Many blockchain wallet transfers that were problematic would have been much less so if there were a one week waiting period. Such a thing is easy to implement in Ethereum and could radically improve security.

Maybe this contract should be called `DelayedClaimable`, and should inherit from `Claimable`

### PR DESCRIPTION
**Pull request by AugustoL**
Extension of the Claimable contract that allows the user to set a timeout for the contract ownership to be claimed using block number.

fixes https://github.com/OpenZeppelin/zeppelin-solidity/issues/64

### CODE REVIEW

**Comment by federicobond** [comment]
I am not a big fan of the `onTime` and `setDelay` names. They sound too generic and I am worried that someone is going to accidentally overwrite them or confuse them with more specific contract functionality.

**Comment by AugustoL** [comment]
@federicobond you are right, what names you suggest?

**Comment by federicobond** [comment]
Can you give me more context on the use case for this kind of logic?

**Comment by maraoz** [comment]
please rebase

**Comment by AugustoL** [comment]
@maraoz No conflicts now, merged with last version on master.

**Comment by federicobond** [comment]
@AugustoL can you squash your commits? Check `git rebase -i`

**Comment by maraoz** [comment]
Squash not required, but good if you do it :)

**Comment by AugustoL** [comment]
@maraoz allow me just this time not to squash, I will do it on next pull request :)

**Comment by maraoz** [comment]
Just reviewed. IMO the modifier `claimBefore` is confusing, as it is only used once and not meant for reuse for child contracts. Do you mind removing it and adding the check in the function instead?

**Comment by AugustoL** [comment]
@maraoz  claimBefore modifier removed :)

**Comment by Qkyrie** [comment]
@AugustoL seems like there's still some merge conflicts in there :)

**Comment by AugustoL** [comment]
@Qkyrie fixed :)

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 dist: trusty
 sudo: false
-group: beta
```
Comment (maraoz) [inline .travis.yml]: why was this change needed?

**Comment by AugustoL** [comment]
@maraoz had to be a mistake, dont know why it was changed, removed.

**Comment by maraoz** [comment]
2 tests failing now :(

**Comment by maraoz** [comment]
@AugustoL I don't know what you did with e6c50b3318435a317206ca2a99311215ed54f851 but it doesn't seem to change any files yet tests are failing now :confused:

**Comment by AugustoL** [comment]
@maraoz because of the change of transfer() to transferOwnership().

**Comment by AugustoL** [comment]
@maraoz ready,

File: contracts/DelayedClaimable.sol [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+contract DelayedClaimable is Ownable, Claimable {
+  uint public claimBeforeBlock;
+
+  function setClaimBefore(uint _claimBeforeBlock) onlyOwner {
+    claimBeforeBlock = _claimBeforeBlock;
+  }
+
+  function claimOwnership() onlyPendingOwner {
+    if (block.number > claimBeforeBlock) throw;
```
Comment (maraoz) [inline contracts/DelayedClaimable.sol]: Just realized there's a big conceptual problem, haha. Sorry I missed this before.

The concept of "delayed claimable" means you have a delay before you can actually claim it. In this case, `claimBeforeBlock` is an expiration timer where the new owner can't claim it after that time has passed.

The logic should be reverse! i.e: `if (block.number < claimAfterBlock) throw;`

**Comment by AugustoL** [comment]
@maraoz understood, but claimBefore can be used too, an auction for example if an item on a contract is sold address A will have X block to claim it, if it is not claimed it will be sold again to the highest bider. What if we set claim before and after flags? where the owner can set a range of blocks for the contract to be claimed?

**Comment by maraoz** [comment]
Yes I like that

**Comment by maraoz** [comment]
LGTM
